Terminal.Gui/Views/ComboBox.cs

```diff
@@ -132,12 +136,49 @@ private void Initialize ()
 			};
 
 			listview.SelectedItemChanged += (ListViewItemEventArgs e) => {
-
+				if (itemClicked) {
+					itemPressed = false;
+					itemClicked = false;
+					SelectedItem = e.Item;
+					if (isShow) {
+						isShow = false;
+						HideList ();
+					}
+					OnOpenSelectedItem ();
+				}
 				if (searchset.Count > 0) {
 					SetValue (searchset [listview.SelectedItem]);
 				}
 			};
 
+			listview.MouseClick += e => {
+				if (isShow && HideDropdownListOnClick && e.MouseEvent.Flags == MouseFlags.Button1Pressed) {
+					itemPressed = true;
+				} else if (itemPressed && isShow && HideDropdownListOnClick && e.MouseEvent.Flags == MouseFlags.Button1Clicked) {
+					if (itemClicked && SelectedItem == listview.SelectedItem) {
+						DoHideList ();
+						e.Handled = true;
+					} else if (itemPressed) {
+						itemClicked = true;
+						Task.Run (async () => {
+							await Task.Delay (100);
```

Comment:
Why is there a delay in closing on select?

Reply:
That was really trickily. The listview `MouseClick` event is invoked before the listview `MouseEvent` is processed. So, I can't rely in the condition `SelectedItem == listview.SelectedItem` itself, because the listview doesn't already processed the change and that why I needed the flags `itemPressed` and `itemClicked`. If another item is selected the `SelectedItemChanged` event will be invoked resetting the flags and hiding the dropdown list. If we click on the same selected item the `SelectedItemChanged` event will not be invoked and we won't know what is the response. That is the reason for the delay to check if the `itemClicked` is true (already processed by the `SelectedItemChanged` event) or false (already processed by the listview `MouseEvent` but without notifying any response).

Reply:
> The listview MouseClick event is invoked before the listview MouseEvent is processed. [...], because the listview doesn't already processed the change 

What about subclassing `ListBox` as a private class e.g. `ComboListBox`.  Then you could override the On Mouse Event method and call `base` method then the custom code to finish up?

Might be more reliable than introducing threading into the equation?

Reply:
I also though about that but I was afraid if it was breaking anything. I'm still doing more unit test because mixing mouse with keyboard doesn't behave as only with mouse. After I have all unit test to guarantee all is fine, maybe I'll do that. Converting to draft to until finishing.

Reply:
I already did what you ask, thanks for the help. Please say something that you think I have to change.